config/routes.rb

```diff
@@ -2820,19 +2820,33 @@
     root :to => "api#index"
 
     # OPTIONS requests for REST API pre-flight checks
-    match '*path' => 'base#handle_options_request', :via => [:options]
+    match '/' => 'base#handle_options_request', :via => :options
+
+    get '/api(/:version)' => 'api#index', :format => 'json', :version => API_VERSION_REGEX
 
     unless defined?(API_ACTIONS)
       API_ACTIONS = {
         :get    => "show",
         :post   => "update",
         :put    => "update",
         :patch  => "update",
-        :delete => "destroy"
+        :delete => "destroy",
+        :options => "options"
       }.freeze
     end
 
+    def action_for_collection(collection_name, verb)
+      "#{collection_name}##{API_ACTIONS[verb]}"
+    end
+
+    def create_api_route(verb, url, action)
+      public_send(verb, url, :to => action)
+    end
+
```

Comment:
why are these added back ?  

Can you squash your commits and git pull upstream master, make sure it's only your diffs that show then repush.  Thanks.